servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpServerBuilder.java

```diff
@@ -30,41 +29,63 @@
 
 import java.net.SocketOption;
 import java.net.StandardSocketOptions;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.function.BooleanSupplier;
 import java.util.function.Predicate;
 import java.util.stream.Stream;
-import javax.annotation.Nullable;
 
 import static io.servicetalk.http.api.BlockingUtils.blockingInvocation;
-import static io.servicetalk.http.api.HttpApiConversions.toStreamingHttpService;
-import static io.servicetalk.http.api.HttpExecutionStrategies.defaultStrategy;
 import static io.servicetalk.http.api.HttpExecutionStrategies.noOffloadsStrategy;
 import static io.servicetalk.http.api.HttpExecutionStrategyInfluencer.defaultStreamingInfluencer;
-import static io.servicetalk.http.api.StrategyInfluencerAwareConversions.toConditionalServiceFilterFactory;
-import static io.servicetalk.transport.api.ConnectionAcceptor.ACCEPT_ALL;
 import static java.util.Objects.requireNonNull;
 
 /**
  * A builder for building HTTP Servers.
  */
-public abstract class HttpServerBuilder {
+public interface HttpServerBuilder {
 
-    @Nullable
-    private ConnectionAcceptorFactory connectionAcceptorFactory;
-    private final List<StreamingHttpServiceFilterFactory> noOffloadServiceFilters = new ArrayList<>();
-    private final List<StreamingHttpServiceFilterFactory> serviceFilters = new ArrayList<>();
-    private HttpExecutionStrategy strategy = defaultStrategy();
-    private boolean drainRequestPayloadBody = true;
+    static StreamingHttpServiceFilterFactory buildFactory(List<StreamingHttpServiceFilterFactory> filters) {
```

Comment:
Consider moving all these `static` utilities to `http-netty` as private or pkg-private utils. Even if users implement this builder interface, it's questionable if they need these utils. We can always make them public later, if there is a good use-case for that.